tests/src/test/fdc3.broadcast.ts

```diff
@@ -13,272 +13,554 @@ export default () =>
+            (context) => {
+              expect(context.type).to.be.equals("fdc3.instrument");
+              resolve();
+            }
+          );
+
+          assert.isObject(listener);
+          expect(typeof listener.unsubscribe).to.be.equals("function");
+        });
+      });
+
+      it("Should only receive the relevant listened context when app B broadcasts multiple contexts to the same app channel", async () => {
+        return new Promise(async (resolve) => {
+          channelsAppContext.joinAppChannel = true;
+          channelsAppContext.contextBroadcasts.contact = true;
+
+          //App A joins app channel
+          const testChannel = await window.fdc3.getOrCreateChannel(
+            "test-channel"
+          );
+
+          //Add context listener to app A
+          listener = await testChannel.addContextListener(
+            "fdc3.instrument",
+            (context) => {
+              expect(context.type).to.be.equals("fdc3.instrument");
+              resolve();
+            }
+          );
+
+          assert.isObject(listener);
+          expect(typeof listener.unsubscribe).to.be.equals("function");
+
+          //App B joins the same app channel as A then broadcasts context
+          await window.fdc3.open("ChannelsApp", channelsAppContext);
+        });
+      });
+
+      it("Should receive multiple contexts when app B broadcasts the listened types to the same app channel", async () => {
+        return new Promise(async (resolve) => {
+          let contextsReceived = 0;
+          channelsAppContext.joinAppChannel = true;
+          channelsAppContext.contextBroadcasts.contact = true;
+
+          //App A joins app channel
+          const testChannel = await window.fdc3.getOrCreateChannel(
+            "test-channel"
+          );
+
+          //Add context listener to app A
+          listener = await testChannel.addContextListener(
+            "fdc3.instrument",
+            (context) => {
+              expect(context.type).to.be.equals("fdc3.instrument");
+              contextsReceived++;
+              checkIfBothContextsReceived();
+            }
+          );
+
+          assert.isObject(listener);
+          expect(typeof listener.unsubscribe).to.be.equals("function");
+
+          //Add a second context listener to app A
+          listener = await testChannel.addContextListener(
+            "fdc3.contact",
+            (context) => {
+              expect(context.type).to.be.equals("fdc3.contact");
+              contextsReceived++;
+              checkIfBothContextsReceived();
+            }
+          );
+
+          assert.isObject(listener);
+          expect(typeof listener.unsubscribe).to.be.equals("function");
+
+          //App B joins the same app channel as A then broadcasts context
+          await window.fdc3.open("ChannelsApp", channelsAppContext);
+
+          function checkIfBothContextsReceived() {
+            contextsReceived++;
+            if (contextsReceived > 1) {
```

Comment:
I'm not sure about this. I'd prefer an expectation for a precise number of contexts or even better storing a list of everything received and then verifying it is as expected